dist/save/index.js

```diff
@@ -0,0 +1,134 @@
+// GCS Cache Post Action - Save caches
+
+// *** Imports ***
+// Node.js dependencies
+const fs = require('fs');
+const path = require('path');
+const md5 = require('md5');
+// GitHub Actions dependencies
+const core = require('@actions/core');
+const exec = require('@actions/exec');
+// Google Cloud client library
+// https://github.com/googleapis/nodejs-storage
+const {Storage} = require('@google-cloud/storage');
+
+// *** Configuration ***
+const uploadDirectory = 'temp';
+const compressionCommand = 'tar -zcf';
+const expandCommand = 'tar -zxf';
+const compressedFileExtension = '.tar.gz';
+const removeCommand = 'rm -r';
+
+// TODO: Move commonly used functions into a module
+// const myModule = require('../module');
+// let val = myModule.hello(); // val is "Hello"
+// console.log(val);
+
+async function run() {
+  try {
+    // Read inputs
+    const noDownload = core.getInput('no-download', { required: false });
+    const cacheKey = core.getInput('cache-key', { required: true });
+    const pathsString = core.getInput('paths', { required: true });
+
+    // Check if provided and read environment variables
+    if (!process.env.IOS_GCS_KEYFILE_BASE64) {
+      throw new Error('No GCS keyfile provided as environment variable');
+    }
+    const gcsKeyfileString = process.env.IOS_GCS_KEYFILE_BASE64;
+    // TODO: Check if it is possible to get the paths of the missing caches from the Main Action via environment variable
+    // const missingCachesPaths = process.env.NOT_FOUND_CACHES;
+
+    // Extract GCS Keyfile
+    const buffer = new Buffer.from(gcsKeyfileString, 'base64');
+    const gcsKeyfile = JSON.parse(buffer.toString('utf8'));
+
+    // Create a GCS client
+    const storage = await new Storage({ credentials: gcsKeyfile });
+    const bucketName = gcsKeyfile.project_id;
+    const bucket = await storage.bucket(bucketName);
+
+    // Get all caches available in Bucket for the given cache key
+    const [availableCaches] = await bucket.getFiles({
+      prefix: cacheKey
+    });
+
+    // List all given path with their corresponding cache references (if exists in the bucket)
+    const cacheObjects = pathsString.split('\n').map((cachePath) => {
+      const cacheFileName = fileNameForCache(cacheKey, cachePath);
+      const cacheReference = availableCaches.find((cache) => {
+        return cache.name == cacheFileName;
+      });
+
+      return [cachePath, cacheReference];
+    });
+
+    const missingCachesMask = cacheObjects.map((cachePath) => {
+      return cachePath[1] === undefined;
+    });
+
+    const filteredCacheObjects = cacheObjects.filter((_, index) => { return missingCachesMask[index] } );
```

Comment:
I am not sure what this code should do. why does it access another array via index? using index of another array is to fragile. Please refactor the code.

```suggestion
    const filteredCacheObjects = cacheObjects.filter((cacheObject) => { 
      const {cacheReference} = cacheObject
      return cacheReference === undefined
    });
```

Reply:
Here I'm working with a boolean mask, which stores a boolean for each given cache object (according their position in the `cacheObjects` array) indicating if they could be found in the GCS bucket or not. 

This mask is used in two purposes:
- [line 67] to check if there is at least one missing cache in the bucket (using the method `.some`).
- [line 75] to filter the array `cacheObjects` and obtaining only those, who has to be saved in the bucket (as shown in the above highlighted lines).

The idea here is to avoid making the same check `cache.cacheReference === undefined` twice.

Do I have to change it anyway?